zilliqa/src/node.rs

```diff
@@ -684,11 +684,7 @@ impl Node {
         let parent = self
             .get_block(header.parent_hash)?
             .ok_or_else(|| anyhow!("missing parent: {}", header.parent_hash))?;
-        let proposer = self
-            .consensus
-            .leader_at_block(&parent, header.view)
-            .unwrap()
-            .public_key;
+        let proposer = self.consensus.leader_at_block(&parent).unwrap().public_key;
```

Comment:
We should use the `leaderAtView()` function of the `Deposit` contract and pass in the `view` argument, otherwise if we execute `leader_at_block()` on the `parent` the view number that determines the randomness in the `Deposit::leader()` function will be the `parent`'s view and not `header.view`.